sdk/src/main/java/com/bugsnag/android/Configuration.java

```diff
@@ -813,8 +814,13 @@ protected void beforeRecordBreadcrumb(@NonNull BeforeRecordBreadcrumb beforeReco
      */
     @Deprecated
     protected boolean inProject(@NonNull String className) {
-        Stacktrace stacktrace = new Stacktrace(new StackTraceElement[]{}, projectPackages);
-        return stacktrace.inProject(className);
+        List<String> packages;
+        if (projectPackages != null) {
```

Comment:
Can't this be simpler?

```java
return Stacktrace.inProject(className, projectPackages);
```

And move the null check/copy into the `inProject` implementation?

```java
static boolean inProject(String className, String[] projectPackages) {
  if (projectPackages == null) {
    return false;
  }
  for (String packageName : Arrays.asList(projectPackages)) {
    if (packageName != null && className.startsWith(packageName)) {
      return true;
    }
  }
  return false;
}
```

Just means that same logic isn't in two separate places. I know it changes the time at which the array is copied to make it immutable/thread safe. If we're worried about that I guess it should get copied as soon as it gets passed in to configuration.

Reply:
Thanks, this makes sense. I've updated the implementation.